test/rpc_test.py

```diff
@@ -243,16 +252,10 @@ def test_register_rpc_backend_and_start_rpc_backend(
     @requires_process_group_agent("PROCESS_GROUP rpc backend specific test, skip")
     @dist_init(setup_model_parallel=False)
     def test_duplicate_name(self):
-        dist.init_process_group(
```

Comment:
Why this test does not require a pg? ProcessGroupAgent would ignore `requires_process_group_agent` and still need the the pg instance to collect names and check if they match, no?

Reply:
Because in this PR, I move the pg initialized inside of `def _init_rpc` instead, `def process_group_init_backend_handler` to be precise.

No, I changed `@requires_process_group_agent` to test agianst test config instead of the current agen instance that has been initialzied.